libs/schema-profunctor/src/Data/Schema.hs

```diff
@@ -316,6 +318,19 @@ array sch = SchemaP (SchemaDoc s) (SchemaIn r) (SchemaOut w)
     s = mkArray (schemaDoc sch)
     w x = A.Array . V.fromList <$> mapM (schemaOut sch) x
 
+nonEmptyArray ::
+  forall ndoc doc a.(HasArray ndoc doc, HasName ndoc, S.HasMinItems doc (Maybe Integer)) =>
+  ValueSchema ndoc a ->
+  ValueSchema doc (NonEmpty a)
+nonEmptyArray sch = SchemaP (SchemaDoc s) (SchemaIn r) (SchemaOut w)
+  where
+    name = maybe "non-empty array" ("non-empty array of " <>) (getName (schemaDoc sch))
+    r = A.withArray (T.unpack name) $ \arr -> case V.toList arr of
+      [] -> A.parseFail "Unexpected empty array found while parsing a NonEmpty"
+      (x : xs) -> mapM (schemaIn sch) (x :| xs)
+    s = mkArray (schemaDoc sch) & S.minItems ?~ 1
+    w xs = A.Array . V.fromList <$> mapM (schemaOut sch) (NonEmpty.toList xs)
+
```

Comment:
It should be possible to use `array` instead of reimplementing its functionality. Something like:
```haskell
nonEmptyArray sch = NonEmpty.toList .= array sch `withParser` check
  where
    check = maybe (fail "Unexpected empty array found while parsing a NonEmpty") pure
          . NonEmpty.nonEmpty
```

Reply:
I couldn't make the swagger work nicely, thought it was ok to have a small amount of duplication. Maybe we can look at it together after lunch to get it to work?